Source/WebCore/rendering/style/StyleBackgroundData.h

```diff
@@ -42,7 +42,8 @@ class StyleBackgroundData : public RefCounted<StyleBackgroundData> {
 
     bool operator==(const StyleBackgroundData&) const;
 
-    bool isEquivalentForPainting(const StyleBackgroundData&, bool currentColorDiffers) const;
+    bool hasVisibleCurrentColor() const;
+    bool isEquivalentForPainting(const StyleBackgroundData*, bool currentColorDiffers) const;
```

Comment:
Not sure why you change the arg to a pointer. Also, I'd prefer all the StyleFoo classes use the same pattern.

Reply:
This is so I can move the pointer comparison in the method

Reply:
Ah I can use arePointingToEqualData() if I have a ref